src/main/grammars/RustParser.bnf

```diff
@@ -448,15 +448,18 @@ TypeParameter ::= OuterAttr* identifier TypeParamBounds? DefaultTypeParameterTyp
 }
 private DefaultTypeParameterType ::= '=' TypeReference { pin = 1 }
 
-ConstParameter ::= OuterAttr* const identifier TypeAscription {
+ConstParameter ::= OuterAttr* const identifier TypeAscription DefaultConstParameter? {
   pin = 2
   implements = [ "org.rust.lang.core.psi.ext.RsGenericParameter"
-                 "org.rust.lang.core.psi.ext.RsOuterAttributeOwner" ]
+                 "org.rust.lang.core.psi.ext.RsOuterAttributeOwner"
+                 "org.rust.lang.core.psi.ext.RsInferenceContextOwner" ]
   mixin = "org.rust.lang.core.psi.ext.RsConstParameterImplMixin"
   stubClass = "org.rust.lang.core.stubs.RsConstParameterStub"
   elementTypeFactory = "org.rust.lang.core.stubs.StubImplementationsKt.factory"
 }
 
+private DefaultConstParameter ::= '=' BlockExpr { pin = 1 }
```

Comment:
We need to improve this part.

As I can see, the following code
```rust
#![feature(const_generics_defaults)]

struct Foo<const N: usize = 1>([i32; N]);

fn main() {}
```

can be compiled successfully by nightly rustc but the plugin parser fails

![image](https://user-images.githubusercontent.com/2539310/132102576-055db9d3-125d-4ee2-9764-97c43e407101.png)

Moreover, the following code is also successfully parsed by rustc

```rust
#![feature(const_generics_defaults)]

struct Foo<const N: usize = 1 + 1>([i32; N]);

fn main() {}
```

because compiler produces semantic error, not parser one
```
error: expressions must be enclosed in braces to be used as const generic arguments
 --> src/main.rs:3:29
  |
3 | struct Foo<const N: usize = 1 + 1>([i32; N]);
  |                             ^^^^^
  |
help: enclose the `const` expression in braces
  |
3 | struct Foo<const N: usize = { 1 + 1 }>([i32; N]);
  |   
```